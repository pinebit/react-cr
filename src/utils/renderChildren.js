import React from 'react';

const renderChildren = (children, props = {}) => {
  const childrenCount = React.Children.count(children);
  const wrapChildren = childrenCount > 1 ||
    typeof children === 'string' ||
    (Array.isArray(children) && childrenCount === 1) ||
    children === undefined;

  if (wrapChildren) {
    return (
      // eslint-disable-next-line
      <div style={props.style || { display: 'inline-block' }} {...props}>
        {children}
      </div>
    );
  } else if (childrenCount) {
    return children;
  }

  return null;
};

export default renderChildren;
