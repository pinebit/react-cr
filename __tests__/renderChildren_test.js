import React from 'react';
import renderChildren from '../src/utils/renderChildren';

it('renderChildren yields null for null', () => {
  const tree = renderChildren(null);
  expect(tree).toBeNull();
});

it('renderChildren yields single node child', () => {
  const children = <h1>Test</h1>;
  const tree = renderChildren(children);
  expect(tree).toBe(children);
});

it('renderChildren yields wrapped single string child', () => {
  const children = 'Test';
  const tree = renderChildren(children);
  expect(tree).toMatchSnapshot();
});

it('renderChildren yields wrapped multiple children', () => {
  const children = React.Children.toArray([
    <h1 key={1}>Test1</h1>,
    <h2 key={2}>Test2</h2>
  ]);
  const tree = renderChildren(children);
  expect(tree).toMatchSnapshot();
});
