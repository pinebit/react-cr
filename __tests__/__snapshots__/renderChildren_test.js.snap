// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderChildren yields wrapped multiple children 1`] = `
<div
  style={
    Object {
      "display": "inline-block",
    }
  }
>
  <h1>
    Test1
  </h1>
  <h2>
    Test2
  </h2>
</div>
`;

exports[`renderChildren yields wrapped single string child 1`] = `
<div
  style={
    Object {
      "display": "inline-block",
    }
  }
>
  Test
</div>
`;
